src/lib.rs

```diff
@@ -191,82 +194,92 @@ impl core::fmt::Display for PReg {
 #[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord, Hash, Default)]
 #[cfg_attr(feature = "enable-serde", derive(Serialize, Deserialize))]
 pub struct PRegSet {
-    bits: [u128; 2],
+    bits: [Bits; Self::LEN],
 }
 
 impl PRegSet {
+    /// The number of bits per element in the internal bit array.
+    const BITS: usize = core::mem::size_of::<Bits>() * 8;
+
+    /// Length of the internal bit array.
+    const LEN: usize = (PReg::NUM_INDEX + Self::BITS - 1) / Self::BITS;
+
     /// Create an empty set.
     pub const fn empty() -> Self {
-        Self { bits: [0; 2] }
+        Self {
+            bits: [0; Self::LEN],
+        }
+    }
+
+    /// Splits the given register index into parts to access the internal bit array.
+    const fn split_index(reg: PReg) -> (usize, usize) {
+        let index = reg.index();
+        (index / Self::BITS, index % Self::BITS)
```

Comment:
It should be optimized via strength reduction but just to be sure in unoptimized builds: could we use shifts and masking here instead? AFAIK, `BITS` will always be a power of two.